docs/build/local-first.md

```diff
@@ -0,0 +1,335 @@
+  sentByMe: true,
+  sentAt: new Date(),
+  contentType: { ...contentType },
+  content: content,
+  isSending: true,
+};
+
+// Save the message to the database and get its ID
+message.id = await db.messages.add(message);
+```
+
+#### Update messages
+
+After a message is sent to the network and you receive the decoded message back, update the original message in the database with the ID of the message on the network and set `isSending` to false.
+
+```jsx
+// Update the message in the database
+await db.messages.update(message.id!, {
+  xmtpID: decodedMessage.id,
+  sentAt: decodedMessage.sent,
+  isSending: false,
+});
+```
+
+#### Check for existing messages
+
+Before saving a received message, check if it already exists in the database. If the message doesn't exist, save it; otherwise, skip the saving process.
+
+```jsx
+// Check if the message already exists in the database
+const existing = await db.messages
+  .where("xmtpID")
+  .equals(decodedMessage.id)
+  .first();
+```
+
+#### Find a conversation
+
+When you need to find a specific conversation in the `conversations` table, search by the `topic` field.
+
+```jsx
+// Find a conversation by topic
+return await db.conversations
+  .where("topic")
+  .equals(stripTopicName(topic))
+  .first();
+```
+
+#### Update a conversation
+
+When a new message is received, update the `updatedAt` timestamp of the related conversation.
+
+```jsx
+// Check if the conversation needs to be updated
+if (conversation && conversation.updatedAt < updatedAt) {
+  // If it does, update the updatedAt timestamp
+  await conversationMutex.runExclusive(async () => {
+    await db.conversations.update(conversation, { updatedAt });
+  });
+}
+```
+
+#### Add conversations
+
+When a new conversation is started, it's first saved to the local database.
+
+```jsx
+// Create a new conversation
+const conversation: Conversation = {
+  /* conversation properties */
+  topic: stripTopicName(xmtpConversation.topic),
+  title: undefined,
+  createdAt: xmtpConversation.createdAt,
+  updatedAt: xmtpConversation.createdAt,
+  isGroup: xmtpConversation.isGroup,
+  peerAddress: xmtpConversation.peerAddress,
+};
+
+// Save the conversation to the database and get its ID
+conversation.id = await db.conversations.add(conversation);
+```
+
+#### Check for existing conversations
+
+Before saving a new conversation, check if it already exists in the database. If the conversation doesn't exist, save it; otherwise, return the existing one.
+
+```jsx
+// Check if the conversation already exists in the database
+const existing = await db.conversations
+  .where("topic")
+  .equals(stripTopicName(xmtpConversation.topic))
+  .first();
+```
+
+### Step 4: Load initial data
+
+<!-- ry - we've removed many of the sections in this section - Use useEffect hook to fetch and save conversations, Use useEffect hook to stream conversations, and Fetch conversations from local DB - does it still make sense as reads now? -->
```

Comment:
Ry - one more question/validation above, please! =)

Reply:
i think so. a lot of this is just understanding how to work with Dexie, and there's tons of official docs for that.